app/services/preserved_object_handler.rb

```diff
@@ -60,7 +60,19 @@ def create
   end
 
   def check_existence
-    # this is just a stub
+    if invalid?
+      handler_results.add_result(PreservedObjectHandlerResults::INVALID_ARGUMENTS, errors.full_messages)
+    elsif PreservedObject.exists?(druid: druid)
+      confirm_version
+      if handler_results.contains_result_code? PreservedObjectHandlerResults::ARG_VERSION_GREATER_THAN_DB_OBJECT
+        update_version_after_validation
```

Comment:
see ticket #410

Reply:
calling the public method 'update_version_after_validation' here is likely contributing to double logging and double db lookups.